ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/terms/LiteralTerms.java

```diff
@@ -209,4 +214,23 @@ public SqlType getSqlType() {
       return SqlTypes.TIMESTAMP;
     }
   }
+
+  public static class IntervalUnitTermImpl implements Term {
+
+    private final TimeUnit value;
+
+    public IntervalUnitTermImpl(final TimeUnit timeUnit) {
+      this.value = timeUnit;
+    }
+
+    @Override
+    public Object getValue(final TermEvaluationContext context) {
+      return value;
+    }
+
+    @Override
+    public SqlType getSqlType() {
+      return null;
```

Comment:
Does this not have a SqlType?  It doesn't seem like it since it's kind of a modifier rather than a data value, but just checking

Reply:
It does not. The only place it can exist is in a function (parser would reject everything else).